### PR TITLE
Add dimension check when parsing WKB GeometryCollections

### PR DESCRIPTION
**Pull request by peterstace**
## Description

Add dimension check when parsing WKB GeometryCollections
    
This ensures that coordinate types are consistent between the geometry collection and its children.

## Check List

Have you:

- Added unit tests? Yes.

- Add cmprefimpl tests? (if appropriate?) N/A

- Updated release notes? (if appropriate?) Yes.

## Related Issue

- https://github.com/peterstace/simplefeatures/issues/397

### CODE REVIEW

File: geom/wkb_test.go [hunk 1]
```diff
@@ -678,3 +679,46 @@ func TestWKBUnmarshalEndianess(t *testing.T) {
...
+					gc.hex +
+					u32One +
+					bigEndian +
+					point.hex +
+					strings.Repeat(f64Zero, point.ct.Dimension())
+				_, err := geom.UnmarshalWKB(hexStringToBytes(t, hex))
+				if gc.ct == point.ct {
+					expectNoErr(t, err)
+				} else {
+					expectErr(t, err)
```
Comment (albertteoh) [inline geom/wkb_test.go:719]: Should we assert that the error type is `mismatchedGeometryCollectionDimsError`?

**Review by albertteoh — APPROVED** [review]

**Comment by peterstace** [comment]
Thanks for reviewing!

File: geom/wkb_test.go (hunk 1 above)
Comment (peterstace) [reply geom/wkb_test.go:719]: Actually.... maybe there is a way to get the best of both worlds. I'll do some experimentation and raise a fix in a new PR if I can get it working.

Comment (peterstace) [reply geom/wkb_test.go:719]: I tried something out in https://github.com/peterstace/simplefeatures/pull/616 (it seems to work pretty well, would love to get your opinion.

**Pull request merged**